src/@types/logger.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * A logger of some sort that will be used to log out warnings, errors
+ * and information based on the enabled level(s)
+ */
+export interface ILogger {
+  /**
+   * A function that logs general information
+   */
+  log?: (message?: string) => void;
+
+  /**
+   * A function that logs warnings
+   */
+  warn?: (message?: string) => void;
+
+  /**
+   * A function that logs error messages
+   */
+  error?: (error?: unknown) => void;
+}
+
+/**
+ * Which logging methods are enabled
+ */
+export type ILevel = 'log' | 'warn' | 'error';
```

Comment:
The internal `ManagedLogger` Only supports these types of logs (and they are optional to implement).
Providing additional levels would indicate to the user that they will receive data in their logger from a method that this library doesn't even use.

The idea with the the Level here being a union is so that you get auto complete like so
![Screen Shot 2020-09-07 at 20 50 55](https://user-images.githubusercontent.com/8206108/92431236-d82eac00-f14b-11ea-9d71-a31f54b04e1b.png)

Reply:
Ah - got it. Seems like a great approach to me!